File: app/src/main/java/com/kyleriedemann/giantbombvideoplayer/Models/Video.java
package com.kyleriedemann.giantbombvideoplayer.Models;

import com.google.gson.annotations.SerializedName;

import org.parceler.Parcel;

/**
 * Created by kyle on 3/31/15.
 */
@Parcel
public class Video {

    @SerializedName("api_detail_url")
    String apiDetailurl;

    @SerializedName("deck")
    String deck;

    @SerializedName("hd_url")
    String hdUrl;

    @SerializedName("high_url")
    String highUrl;

    @SerializedName("low_url")
    String lowUrl;

    @SerializedName("id")
    int id;

    @SerializedName("length_seconds")
    int lengthSeconds;

    @SerializedName("name")
    String name;

    @SerializedName("publish_date")
    String publishDate;

    @SerializedName("site_detail_url")
    String siteDetailUrl;

    @SerializedName("url")
    String ulr;

    @SerializedName("image")
    Image videoImage;

    @SerializedName("user")
    String user;

    @SerializedName("video_type")
    String videoType;

    @SerializedName("youtube_id")
    String youtubeId;

    public Video() {
    }

    public String getApiDetailurl() {
        return apiDetailurl;
    }

    public void setApiDetailurl(String apiDetailurl) {
        this.apiDetailurl = apiDetailurl;
    }

    public String getDeck() {
        return deck;
    }

    public void setDeck(String deck) {
        this.deck = deck;
    }

    public String getHdUrl() {
        return hdUrl;
    }

    public void setHdUrl(String hdUrl) {
        this.hdUrl = hdUrl;
    }

    public String getHighUrl() {
        return highUrl;
    }

    public void setHighUrl(String highUrl) {
        this.highUrl = highUrl;
    }

    public String getLowUrl() {
        return lowUrl;
    }

    public void setLowUrl(String lowUrl) {
        this.lowUrl = lowUrl;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getLengthSeconds() {
        return lengthSeconds;
    }

    public void setLengthSeconds(int lengthSeconds) {
        this.lengthSeconds = lengthSeconds;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPublishDate() {
        return publishDate;
    }

    public void setPublishDate(String publishDate) {
        this.publishDate = publishDate;
    }

    public String getSiteDetailUrl() {
        return siteDetailUrl;
    }

    public void setSiteDetailUrl(String siteDetailUrl) {
        this.siteDetailUrl = siteDetailUrl;
    }

    public String getUlr() {
        return ulr;
    }

    public void setUlr(String ulr) {
        this.ulr = ulr;
    }

    public Image getVideoImage() {
        return videoImage;
    }

    public void setVideoImage(Image videoImage) {
        this.videoImage = videoImage;
    }

    public String getUser() {
        return user;
    }

    public void setUser(String user) {
        this.user = user;
    }

    public String getVideoType() {
        return videoType;
    }

    public void setVideoType(String videoType) {
        this.videoType = videoType;
    }

    public String getYoutubeId() {
        return youtubeId;
    }

    public void setYoutubeId(String youtubeId) {
        this.youtubeId = youtubeId;
    }

    @Override
    public String toString() {
        return "Video{" +
                "apiDetailurl='" + apiDetailurl + '\'' +
                ", deck='" + deck + '\'' +
                ", hdUrl='" + hdUrl + '\'' +
                ", highUrl='" + highUrl + '\'' +
                ", lowUrl='" + lowUrl + '\'' +
                ", id=" + id +
                ", lengthSeconds=" + lengthSeconds +
                ", name='" + name + '\'' +
                ", publishDate='" + publishDate + '\'' +
                ", siteDetailUrl='" + siteDetailUrl + '\'' +
                ", ulr='" + ulr + '\'' +
                ", videoImage=" + videoImage +
                ", user='" + user + '\'' +
                ", videoType='" + videoType + '\'' +
                ", youtubeId='" + youtubeId + '\'' +
                '}';
    }
}

File: app/src/main/java/com/kyleriedemann/giantbombvideoplayer/video/models/Video.kt
package com.kyleriedemann.giantbombvideoplayer.video.models

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Embedded
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import com.kyleriedemann.giantbombvideoplayer.video.list.VideoCard

@Entity(tableName = "video")
data class Video(
        @PrimaryKey @SerializedName("id") var id: Int = 0,
        @ColumnInfo(name = "api_detail_url") @SerializedName("api_detail_url") var apiDetailurl: String? = "",
        @ColumnInfo(name = "deck") @SerializedName("deck") var deck: String? = "",
        @ColumnInfo(name = "hd_url") @SerializedName("hd_url") var hdUrl: String? = "",
        @ColumnInfo(name = "high_url") @SerializedName("high_url") var highUrl: String? = "",
        @ColumnInfo(name = "low_url") @SerializedName("low_url") var lowUrl: String? = "",
        @ColumnInfo(name = "length_seconds") @SerializedName("length_seconds") var lengthSeconds: Int = 0,
        @ColumnInfo(name = "name") @SerializedName("name") var name: String? = "",
        @ColumnInfo(name = "publish_date") @SerializedName("publish_date") var publishDate: String? = "",
        @ColumnInfo(name = "site_detail_url") @SerializedName("site_detail_url") var siteDetailUrl: String? = "",
        @ColumnInfo(name = "url") @SerializedName("url") var ulr: String? = "",
        @Embedded @SerializedName("image") var videoImage: Image = Image("", "", "", "", "", "", ""),
        @ColumnInfo(name = "user") @SerializedName("user") var user: String? = "",
        @ColumnInfo(name = "video_type") @SerializedName("video_type") var videoType: String? = "",
        @ColumnInfo(name = "youtube_id") @SerializedName("youtube_id") var youtubeId: String? = "",
        @ColumnInfo(name = "saved_time") @SerializedName("saved_time") var savedTime: Long = 0
)

fun Video?.toVideoItem(): VideoCard {
    return VideoCard(this)
}